test/e2e/main_test.go

```diff
@@ -2,10 +2,79 @@ package e2e
 	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/pkg/errors"
+	"k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
 )
 
+// Expose this as a flag.
+var noOfNodes = 1
+
+// globalContext holds the information that we want to use across the test suites.
+// If you want to move item here make sure that
+// 1.) It is needed across test suites
+// 2.) You're responsible for checking if the field is stale or not. Any field
+//     in this struct is not guaranteed to be latest from the apiserver.
+type globalContext struct {
+	// windowsVMs has the slices of WindowsVMs that are created by WMCO. These are
+	// populated from the secrets generated by WMCO.
+	windowsVMs []types.WindowsVM
+	// numberOfNodes to be used for the test suite.
+	numberOfNodes int
+}
+
+// testContext holds the information related to the entire test suite. This data structure
+// should be instantiated by every test suite, so that we can update the test context to be
+// passed around to get the information which was created within the test suite. For example,
+// if the create test suite creates a Windows Node object, the node object and other related
+// information should be easily accessible by other methods within the same test suite.
+// Some of the fields we have here can be exposed by via flags
+// to the test suite.
+type testContext struct {
+	// namespace holds the namespace of
+	namespace string
+	// osdkTestCtx holds the information related to operator sdk framework's test Context
+	osdkTestCtx *framework.TestCtx
+	// nodes holds the list of nodes
+	nodes []v1.Node
+	// credentials to be used to the node.
+	credentials []tracker.Credentials
+	// retryInterval to check for existence of resource in kube api server
+	retryInterval time.Duration
+	// timeout to terminate checking for the existence of resource in kube apiserver
+	timeout time.Duration
+	// kubeclient holds the kube client
+	kubeclient kubernetes.Interface
+	// tracketConfigmap holds the information related to the configmap created by controller
+	trackerConfigMap *v1.ConfigMap
+}
+
+// NewTestContext returns a new test context to be used by every test.
+func NewTestContext(t *testing.T) (*testContext, error) {
+	fmwkTestContext := framework.NewTestCtx(t)
+	namespace, err := fmwkTestContext.GetNamespace()
+	if err != nil {
+		return nil, errors.Wrap(err, "test context instantiation failed")
+	}
+	// number of nodes, retry interval and timeout should come from user-input flags
+	return &testContext{osdkTestCtx: fmwkTestContext, retryInterval: time.Second * 5,
+		timeout: time.Minute * 15, kubeclient: framework.Global.KubeClient, nodes: make([]v1.Node, 0, gc.numberOfNodes),
+		namespace: namespace}, nil
+}
+
+// cleanup cleans up the test context
+func (tc *testContext) cleanup() {
+	tc.osdkTestCtx.Cleanup()
+}
+
+// gc to hold the global context across the test suites.
+var gc = globalContext{numberOfNodes: 1, windowsVMs: make([]types.WindowsVM, 0, 1)}
```

Comment:
- Please use the global variable `nrNodes` in the `make` in place of `1`.
- Please group global variable together